### PR TITLE
Add a blog post on Distro Puppet acceptance testing

### PR DESCRIPTION
**Pull request by ekohl**

### CODE REVIEW

File: _posts/2023-10-18-distro-puppet-acceptance-tests.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+category: changelog
+---
+
+Vox Pupuli's CI now supports testing using distro Puppet versions.
+At the moment this is used for ArchLinux, Debian 12 and Fedora 37 & 38.
+
+Within Vox Pupuli we maintain a lot of modules.
+Keeping those healthy is a challenge and having working acceptance tests makes that feasible.
+For each operating system we need two things: a working (container) image and Puppet.
+The former is generally easy (except for operating systems with licenses), but for the we need some system packages.
```
Comment (bastelfreak) [inline _posts/2023-10-18-distro-puppet-acceptance-tests.md:15]: `for the we`? Do you mean `for the latter`?

**Comment by bschonec** [comment]
Thank you.  This does make it easier for an amateur like me to get started on the journey of contributing to Voxpupuli.

File: _posts/2023-10-18-distro-puppet-acceptance-tests.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+Until recently we've relied Puppet's [RPM](https://yum.puppet.com/) and [deb](https://apt.puppet.com/) packages.
```
Comment (bastelfreak) [inline _posts/2023-10-18-distro-puppet-acceptance-tests.md:16]: `we've relied Puppet's` or `we've relied on Puppet's`?

File: _posts/2023-10-18-distro-puppet-acceptance-tests.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+This matches how most people consume Puppet, but Puppet's coverage of operating systems is incomplete.
+There are no packages for Archlinux and on the distributions that are supported, it doesn't have complete coverage.
+Now our tooling is enhanced to use the distribution's native packages.
+With this in place, it's possible to start supporting the recently released Debian 12.
+
+It's also a good chance to dive a bit deeper in how this is done.
+Back in February I gave [a talk](https://cfp.cfgmgmtcamp.org/2023/talk/DWBWFX/) on this, but there was a lot of information in there and a written form may useful.
```
Comment (bastelfreak) [inline _posts/2023-10-18-distro-puppet-acceptance-tests.md:23]: `written form may useful` or `written form may be useful`?

File: _posts/2023-10-18-distro-puppet-acceptance-tests.md (hunk 2 above)
Comment (bschonec) [reply _posts/2023-10-18-distro-puppet-acceptance-tests.md:16]: Also, line 15, "but for the we"

**Comment by ekohl** [comment]
As a side note: we should not publish this until our modulesync has been updated with https://github.com/voxpupuli/modulesync_config/pull/855.

File: _posts/2023-10-18-distro-puppet-acceptance-tests.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+I'm not going to cover everything from my talk, but only focus on the acceptance test implementation.
+It heavily relies on dynamically generated matrices in GitHub Actions, which I've [written](https://partial.solutions/2023/advanced-matrix-testing-with-github-actions.html) about before.
+To generate this matrix, [puppet_metadata](https://github.com/voxpupuli/puppet_metadata) is used.
+
+How does `puppet_metadata` generate this matrix?
+It parses `metadata.json` and builds a [list of supported operating systems](https://github.com/voxpupuli/puppet_metadata/blob/ff57b5e1fbc5ea09c6d4e5ca91e8e983f56d5585/lib/puppet_metadata/github_actions.rb#L50-L75).
+For every operating system it iterates the OS releases (except for rolling releases like Archlinux and Gentoo).
+For each release, it iterates the supported Puppet versions (also derived from `metadata.json`) and checks if it can find a Puppet version.
+If there's an AIO build (determined using [PuppetMetadata::AIO](https://github.com/voxpupuli/puppet_metadata/blob/master/lib/puppet_metadata/aio.rb)) then that's preferred.
```
Comment (kenyon) [inline _posts/2023-10-18-distro-puppet-acceptance-tests.md:32]:
```suggestion
If there's an AIO (all-in-one) build (determined using [PuppetMetadata::AIO](https://github.com/voxpupuli/puppet_metadata/blob/master/lib/puppet_metadata/aio.rb)) then that's preferred.
```

**Review by kenyon — APPROVED** [review]

**Comment by ekohl** [comment]
Addressed the last issue. Also rebased and updated the date to $today.

**Review by bastelfreak — APPROVED** [review]

**Pull request merged**